src/plugins/discover/public/application/components/discover_grid/discover_grid_document_selection.tsx

```diff
@@ -0,0 +1,169 @@
+      isFilterActive ? (
+        <EuiContextMenuItem
+          data-test-subj="dscGridShowAllDocuments"
+          key="showAllDocuments"
+          icon="eye"
+          onClick={() => {
+            setIsSelectionPopoverOpen(false);
+            setIsFilterActive(false);
+          }}
+        >
+          <FormattedMessage id="discover.showAllDocuments" defaultMessage="Show all documents" />
+        </EuiContextMenuItem>
+      ) : (
+        <EuiContextMenuItem
+          data-test-subj="dscGridShowSelectedDocuments"
+          key="showSelectedDocuments"
+          icon="eye"
+          onClick={() => {
+            setIsSelectionPopoverOpen(false);
+            setIsFilterActive(true);
+          }}
+        >
+          <FormattedMessage
+            id="discover.showSelectedDocumentsOnly"
+            defaultMessage="Show selected documents only"
+          />
+        </EuiContextMenuItem>
+      ),
+
+      <EuiContextMenuItem
+        data-test-subj="dscGridClearSelectedDocuments"
+        key="clearSelection"
+        icon="cross"
+        onClick={() => {
+          setIsSelectionPopoverOpen(false);
+          setSelectedDocs([]);
+          setIsFilterActive(false);
+        }}
+      >
+        <FormattedMessage id="discover.clearSelection" defaultMessage="Clear selection" />
+      </EuiContextMenuItem>,
+      <EuiCopy
+        key="copyJsonWrapper"
+        data-test-subj="dscGridCopySelectedDocumentsJSON"
+        textToCopy={
+          !rows ? '' : JSON.stringify(rows.filter((row) => selectedDocs.includes(getDocId(row))))
+        }
+      >
+        {(copy) => (
+          <EuiContextMenuItem key="copyJSON" icon="copyClipboard" onClick={copy}>
+            <FormattedMessage
+              id="discover.copyToClipboardJSON"
+              defaultMessage="Copy documents to clipboard (JSON)"
+            />
+          </EuiContextMenuItem>
+        )}
+      </EuiCopy>,
+    ];
+  }, [
+    isFilterActive,
+    rows,
+    selectedDocs,
+    setIsFilterActive,
+    setIsSelectionPopoverOpen,
+    setSelectedDocs,
+  ]);
+
+  return (
+    <EuiPopover
+      closePopover={() => setIsSelectionPopoverOpen(false)}
+      isOpen={isSelectionPopoverOpen}
+      button={
+        <EuiButtonEmpty
+          size="xs"
+          color="text"
+          iconType="documents"
+          onClick={() => setIsSelectionPopoverOpen(true)}
+          data-selected-documents={selectedDocs.length}
+          data-test-subj="dscGridSelectionBtn"
+          isSelected={isFilterActive}
+          className={classNames({
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            euiDataGrid__controlBtn: true,
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            'euiDataGrid__controlBtn--active': isFilterActive,
+          })}
+        >
+          <FormattedMessage
+            id="discover.selectedDocumentsNumber"
+            defaultMessage="{nr} documents selected"
```

Comment:
Nit: for localization purposes, better format would be:
`documents selected: {nr}`

Reply:
true! thx, will do!

Reply:
@majagrubic I do agree that i18n would be better, however it was chosen for consistency, because the grid days "1 field sorted"

<img width="312" alt="Bildschirmfoto 2021-04-13 um 08 33 35" src="https://user-images.githubusercontent.com/463851/114507462-71b6c280-9c33-11eb-9b83-aac55bb2841f.png">
<img width="322" alt="Bildschirmfoto 2021-04-13 um 08 33 01" src="https://user-images.githubusercontent.com/463851/114507474-74191c80-9c33-11eb-9054-47b2cbca933c.png">
@majagrubic so I think we should go for consistency in this case